docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -175,17 +185,20 @@ index                                 prirep shard store
 
 [discrete]
 [[shard-count-recommendation]]
-==== Aim for 20 shards or fewer per GB of heap memory
-
-The number of shards a data node can hold is proportional to the node's heap
-memory. For example, a node with 30GB of heap memory should have at most 600
-shards. The further below this limit you can keep your nodes, the better. If
-you find your nodes exceeding more than 20 shards per GB, consider adding
-another node.
-
-Some system indices for {enterprise-search-ref}/index.html[Enterprise Search]
-are nearly empty and rarely used. Due to their low overhead, you shouldn't
-count shards for these indices toward a node's shard limit.
+==== Aim for 3000 indices or fewer per GB of heap memory on master nodes
+
+The number of indices a master node can manage is proportional to the node's
+heap memory. The exact amount of heap memory each additional index requires
+depends on various factors such as the size of the mapping and the number of
+shards per index. Where possible, the master node will deduplicate the mapping
+metadata across indices with identical mappings. Mapping metadata can be
+deduplicated across indices that are created from the same index template as
```

Comment:
We could pretty easily since mapping stats computation uses deduplication internally these days.

Reply:
:+1: should we do that before making these changes? It'd be a lot simpler to say "check this API" rather than just trying to explain the concepts in words.

Reply:
I'd say we can adjust the docs here a little more when making those changes? IMO these changes still get us to a better spot than where we currently are and I'm not sure we should delay this that much more?

Reply:
Ok, I opened #86639 to track this.